test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperCheckboxTest.cs

```diff
@@ -128,6 +128,26 @@ public void CheckBoxCheckedWithOnlyName_GeneratesExpectedValue()
         }
 
         [Fact]
+        public void CheckBox_WithCanRenderAtEndOfFormSet_DoesNotGenerateInlineHiddenTag()
+        {
+            // Arrange
+            // Mono issue - https://github.com/aspnet/External/issues/19
+            var expected = PlatformNormalizer.NormalizeContent(
+                @"<input checked=""HtmlEncode[[checked]]"" data-val=""HtmlEncode[[true]]"" " +
+                @"data-val-required=""HtmlEncode[[The Boolean field is required.]]"" id=""HtmlEncode[[Property1]]"" " +
+                @"name=""HtmlEncode[[Property1]]"" type=""HtmlEncode[[checkbox]]"" " +
+                @"value=""HtmlEncode[[true]]"" />");
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(GetTestModelViewData());
+            helper.ViewContext.FormContext.CanRenderAtEndOfForm = true;
+
+            // Act
+            var html = helper.CheckBox("Property1", isChecked: true, htmlAttributes: null);
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(html));
```

Comment:
Since this is a unit test, add assertions about `FormContext.HasRenderAtEndOfFormContent` and the `RenderAtEndOfFormContent` itself.